xknx/telegram/apci.py

```diff
@@ -158,6 +163,14 @@ def from_knx(cls, raw: bytes) -> APCI:
         if service == APCIService.ADC_READ.value:
             return ADCRead.from_knx(raw)
         if service == APCIService.ADC_RESPONSE.value:
+            if apci == APCIService.MEMORY_EXTENDED_WRITE.value:
```

Comment:
unindent it. 
`service` cant be `ADC_RESPONSE` (0x1c0) AND `MEMORY_EXTENDED_WRITE` (0x1fb) at the same time. Just append your `if`s.

Reply:
oh, sorry. I mussed the `if apci ==` in Line 166. I was thinking this was `service` too. 
Maybe have a look at `APCIExtendedService`. I think it would make sense to move the constant definitions for `MEMORY_EXTENDED` to its own Enum since it is not used to match  the first 4 bit of APCI, but the latter 6.

Then, yes, your approach looks fine. It looks like the specifications aren't perfectly clear at this point what `A_ADC_Response-PDU` last 6 bit may or may not contain.
Maybe add a comment before Line 174 `return ADCResponse.from_knx(raw)`.